Fix: Indent Ignore Variable Declaration init operator (fixes #8546)

soda0289 opened the pull request:
**What is the purpose of this pull request? (put an "X" next to item)**

[X ] Bug fix (fixes #8546 )

**What changes did you make? (Give an overview)**
Allow the VariableDeclarator node listener to ignore the operator as well as the initial value.


**Is there anything you'd like reviewers to focus on?**
I tried to match the behavior of variable assignment but might have missed something. The following are now all valid:
```js
const aLongName 
    = 1,
  bLongName
    = 2;
```
```js
const aLongName 
    =
  1,
  bLongName
     =
     2;
```

not-an-aardvark commented on lib/rules/indent.js:
```diff
@@ -1011,6 +1011,11 @@ module.exports = {
 
             VariableDeclarator(node) {
                 if (node.init) {
+                    const operator = sourceCode.getTokenBefore(node.init);
```
I think this will work incorrectly if `node.init` is parenthesized. Using `sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken)` should resolve that.

not-an-aardvark commented on lib/rules/indent.js:
```diff
@@ -1011,6 +1011,11 @@ module.exports = {
                 if (node.init) {
+                    const operator = sourceCode.getTokenBefore(node.init);
+
+                    if (operator.type === "Punctuator" && operator.value === "=" && tokenInfo.isFirstTokenOfLine(operator)) {
```
Why is it necessary to check that the operator is an equals sign?

not-an-aardvark commented on lib/rules/indent.js:
```diff
@@ -1011,6 +1011,11 @@ module.exports = {
+                    const operator = sourceCode.getTokenBefore(node.init);
+
+                    if (operator.type === "Punctuator" && operator.value === "=" && tokenInfo.isFirstTokenOfLine(operator)) {
+                        offsets.ignoreToken(operator);
```
`offsets.ignoreToken` is a no-op if a token is not the first token of its line, so it's not necessary to check for that in the `if` statement.

soda0289 commented:
@not-an-aardvark Thanks for the review. I can't reply to your comments, some JS isn't loading, but I made the following changes:

- I used `astUtils.isNotOpeningParenToken()` to ensure the previous token is not a parenthese. Also added a test for this case
- Removed the check to ensure it was an equal operator. I think I was concerned it could be a comma or some other token but cannot think of an example
- Removed the first token of line since it is not needed

not-an-aardvark commented on lib/rules/indent.js:
```diff
@@ -1039,6 +1039,11 @@ module.exports = {
                 if (node.init) {
+                    const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
+
+                    if (equalOperator) {
```
Will this condition ever be false?

soda0289 replied:
I thought that since getTokenBefore() can return null, and ignoreToken() cannot handle null values, I should have a check. In this case it should always return the equal operator.

I will remove the if statement to clean this up.

not-an-aardvark replied:
I suppose that makes sense, but if that happens, then some invariant has been violated and it's likely that the rule will do the wrong thing anyway. So I think it's better to avoid unnecessary checks unless we know that they're necessary in a particular case.

soda0289 commented:
Removed the if statement complete. Thanks.
